src/stores/useProductStore.ts

```diff
@@ -0,0 +1,39 @@
+import { create } from 'zustand';
+import { DataStatus, Product } from '../types/products';
+
+type ProductStoreState = {
+  products: Product[];
+  status: DataStatus;
+  total: number | undefined;
+};
+
+type ProductStoreActions = {
+  setProducts: (
+    nextProducts:
+      | ProductStoreState['products']
+      | ((currentProducts: ProductStoreState['products']) => ProductStoreState['products']),
+  ) => void;
+  setStatus: (nextStatus: ProductStoreState['status']) => void;
+  setTotal: (nextTotal: ProductStoreState['total']) => void;
+  getProductById: (id: Product['id']) => Product | undefined;
+};
+
+type ProductStore = ProductStoreState & ProductStoreActions;
+
+export const useProductStore = create<ProductStore>((set, get) => ({
+  products: [],
+  status: 'loading',
+  total: undefined,
+  setProducts: (nextProducts) => {
+    set((state) => ({
+      products: typeof nextProducts === 'function' ? nextProducts(state.products) : nextProducts,
+    }));
+  },
+  setStatus: (nextStatus) => {
+    set(() => ({ status: nextStatus }));
```

Comment:
you can pass the object if you don't need the previous state
```suggestion
    set({ status: nextStatus });
```

Reply:
Ok! thank you